src/test/java/org/springframework/data/couchbase/core/mapping/MappingCouchbaseConverterTests.java

```diff
@@ -788,89 +786,145 @@ public DateEntity(Date created, Calendar modified, LocalDateTime deleted) {
 	}
 
 	@Test
-	void idTest00() { // id does not get set
+	void idNotGenerated() {
+		class Entity {
+			public static final String ID = "mockid";
+			@Id private String id = ID;
+		}
+		Entity entity = new Entity();
 		CouchbaseDocument converted = new CouchbaseDocument();
-		IdTest00Entity entity = new IdTest00Entity();
 		converter.write(entity, converted);
 		Map<String, Object> result = converted.export();
-		assertThat(converted.getId()).isEqualTo(entity.getId());
+		assertThat(converted.getId()).isEqualTo(entity.id);
 	}
 
-	public static class IdTest00Entity { // id does not get set
-		public static final String ID = "mockid";
-		@Id private String id = ID;
-
-		String getId() {
-			return id;
+	@Test
+	void idIsGeneratedString() {
+		class Entity {
+			@GeneratedValue(strategy = GenerationStrategy.UNIQUE) @Id String id;
 		}
+		Entity entity = new Entity();
+		CouchbaseDocument converted = new CouchbaseDocument();
+		converter.write(entity, converted);
+		Map<String, Object> result = converted.export();
+		assertThat(converted.getId()).isEqualTo(entity.id);
 	}
 
 	@Test
-	void idTest02() { // id is String
+	void idIsGeneratedUUID() {
+		class Entity {
+			@GeneratedValue(strategy = GenerationStrategy.UNIQUE) @Id UUID id;
+		}
+		Entity entity = new Entity();
 		CouchbaseDocument converted = new CouchbaseDocument();
-		IdTest02Entity entity = new IdTest02Entity();
 		converter.write(entity, converted);
-		Map<String, Object> result = converted.export();
-		assertThat(converted.getId()).isEqualTo(entity.getId());
+		assertThat(converted.getId()).isEqualTo(entity.id.toString());
+		System.out.println(entity.id);
```

Comment:
Looks like there are leftover printlns in addition to the assertions?

Reply:
fixed.